src/osd/scrubber/scrub_backend.h

```diff
@@ -113,10 +114,17 @@ struct ScrubBeListener {
 
 struct SnapMapperAccessor {
   virtual int get_snaps(const hobject_t& hoid,
-                        std::set<snapid_t>* snaps_set) const = 0;
+			std::set<snapid_t>* snaps_set) const = 0;
+  virtual tl::expected<std::set<snapid_t>, int> get_snaps(
+    const hobject_t& hoid) const = 0;
   virtual ~SnapMapperAccessor() = default;
 };
 
+/*
+return tl::unexpected(info->last_error);
+
+*/
+
 enum class snap_mapper_op_t {
   add,
   update,
```

Comment:
We already have those in mapper access, can it be removed from here? (`snap_mapper_op_t` and `snap_mapper_fix_t`)